log.go

```diff
@@ -394,5 +404,26 @@ func (l *Logger) should(lvl Level) bool {
 	if l.sampler != nil && !samplingDisabled() {
 		return l.sampler.Sample(lvl)
 	}
+
+	if l.section != "" {
+		enabledSections := SectionsProviderFunc()
+
+		for _, entry := range strings.Split(enabledSections, ",") {
+			section := strings.Split(entry, "=")
+			sectionName := section[0]
+			sectionLevel := NoLevel
+
+			if sectionName != "" {
+				if len(section) > 1 {
+					sectionLevel, _ = ParseLevel(section[1])
+				}
+				if matched, _ := regexp.MatchString(sectionName, l.section); matched {
```

Comment:
I would prefer to limit the pattern matching capabilities to a simple wildcard and avoid the use of regexp.